examples/tutorials/hybrid_demucs_tutorial.py

```diff
@@ -0,0 +1,360 @@
+    overlap=overlap,
+)[0]
+sources = sources * ref.std() + ref.mean()
+
+sources_list = model.sources
+sources = list(sources)
+
+audios = dict(zip(sources_list, sources))
+
+
+######################################################################
+# 5.1 Separate Track
+# ^^^^^^^^^^^^^^^^^^
+# 
+# The default set of pretrained weights that has been loaded has 4 sources
+# that it is separated into: drums, bass, other, and vocals in that order.
+# They have been stored into the dict “audios” and therefore can be
+# accessed there. For the four sources, there is a separate cell for each,
+# that will create the audio, the spectrogram graph, and also calculate
+# the SDR score. SDR is the signal-to-distortion
+# ratio, essentially a representation to the “quality” of an audio track.
+# 
+
+N_FFT = 4096
+N_HOP = 4
+stft = torchaudio.transforms.Spectrogram(
+    n_fft=N_FFT,
+    hop_length=N_HOP,
+    power=None,
+)
+
+
+######################################################################
+# 5.2 Audio Segmenting and Processing
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# 
+# Below is the processing steps and segmenting 5 seconds of the tracks in
+# order to feed into the spectrogram and to caclulate the respective SDR
+# scores.
+# 
+
+drums_original = download_asset("tutorial-assets/hdemucs_drums_segment.wav")
+bass_original = download_asset("tutorial-assets/hdemucs_bass_segment.wav")
+other_original = download_asset("tutorial-assets/hdemucs_other_segment.wav")
+vocals_original = download_asset("tutorial-assets/hdemucs_vocals_segment.wav")
+
+track = store
+mix_spec = track[:, 150*sample_rate: 155*sample_rate]
+
+drums_spec = audios["drums"][:, 150*sample_rate: 155*sample_rate]
+drums, sample_rate = torchaudio.load(drums_original)
+
+bass_spec = audios["bass"][:, 150*sample_rate: 155*sample_rate]
+bass, sample_rate = torchaudio.load(bass_original)
+
+other_spec = audios["other"][:, 150*sample_rate: 155*sample_rate]
+other, sample_rate = torchaudio.load(other_original)
+
+vocals_spec = audios["vocals"][:, 150*sample_rate: 155*sample_rate]
+vocals, sample_rate = torchaudio.load(vocals_original)
+
+
+######################################################################
+# 5.3 Spectrograms and Audio
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^
+# 
+# In the next 5 cells, you can see the spectrograms with the respective
+# audios. The audios can be clearly visualized using the spectrogram.
+# 
+# The mixture clip comes from the original track, and the remaining 4
+# tracks are the model output
+# 
+
+# Mixture Clip
+plot_spectrogram(stft(mix_spec)[0], "Spectrogram Mixture")
+Audio(mix_spec, rate=sample_rate)
+
+
+######################################################################
+# Drums SDR, Spectrogram, and Audio
+# 
+
+# Drums Clip
+print("SDR score is:", separation.bss_eval_sources(drums.detach().numpy(), drums_spec.detach().numpy())[0].mean())
+plot_spectrogram(stft(drums_spec)[0], "Spectrogram Drums")
+Audio(drums_spec, rate=sample_rate)
```

Comment:
The logic expressed here is repeated multiple times. can you define a helper function and re-use it?